utils/denoDoc.ts

```diff
@@ -0,0 +1,363 @@
+ */
+const get$idFromImportUrl = (importUrl: string) => {
+  // TODO: Improve this to not include live's version when using deno.lan
+  // E.g: live@0.1.16-functions/std/commerce/types/ProductList.ts
+  return "live" + importUrl?.split("/live")[1];
+};
+
+export async function readDenoDocJson(filePath: string) {
+  const { output: rawOutput } = await exec(`deno doc ${filePath} --json`, {
+    output: OutputMode.Capture,
+  });
+  const denoDocOutput = JSON.parse(rawOutput);
+  return denoDocOutput;
+}
+
+export function getFunctionOutputSchemaFromDocs(
+  docs: DenoDocResponse[],
+  entityName: string
+) {
+  const defaultFunctionExport = docs.find(
+    ({ name, kind }) => name === "default" && kind === "variable"
+  );
+
+  if (!defaultFunctionExport) {
+    console.log(
+      `${entityName} should have a named function as the default export. Check the docs here: #TODO`
+    );
+  }
+
+  const functionTsType = defaultFunctionExport?.variableDef?.tsType;
+  const functionTypeName = functionTsType?.repr; // E.g: LoaderFunction
+
+  const VALID_FUNCTION_TYPES = ["LoaderFunction"];
+
+  if (
+    !functionTypeName ||
+    !functionTsType ||
+    !VALID_FUNCTION_TYPES.includes(functionTypeName)
+  ) {
+    console.log(
+      `${entityName} should export a function with one of Live's provided types: LoaderFunction.`
+    );
+
+    return null;
+  }
+
+  const outputTypeName = functionTsType?.typeRef?.typeParams?.map(
+    ({ repr }) => repr
+  )[1]; // E.g: Product
+
+  if (!outputTypeName && functionTypeName === "LoaderFunction") {
+    console.log(
+      `${entityName} should specify its return type like this: LoaderFunction<Props, Product>, where Product is the return type.`
+    );
+    return null;
+  }
+
+  const outputTypeUrl = docs.find(
+    ({ kind, name }) => kind === "import" && name === outputTypeName
+  )?.importDef?.src; // E.g: file:///Users/lucis/deco/live/std/commerce/types/Product.ts
+
+  if (!outputTypeUrl) {
+    console.log(
+      `Couldn't find import for output type '${outputTypeName}' in ${entityName}.`
+    );
+    return null;
+  }
+
+  const typeId = get$idFromImportUrl(outputTypeUrl);
+
+  const outputSchema: JSONSchema7 = {
+    type: "object",
+    properties: {
+      data: {
+        $id: typeId,
+      },
+    },
+    // Technically, this function might return additional data (like headers and status), but
+    // they don't matter for the schema now
+    additionalProperties: true,
+  };
+
+  return outputSchema;
+}
+
+export function getInputSchemaFromDocs(
+  docs: DenoDocResponse[],
+  entityName?: string
+): JSONSchema7 | null {
+  const propsExport = docs.find(({ name }) => name === "Props");
+
+  if (!propsExport) {
+    entityName &&
+      console.log(
+        `${entityName} doesn't export a Props interface definition, couldn't extract schema.`
+      );
+    return null;
+  }
+
+  const componentName = propsExport.location.filename
+    .split("/")
+    .pop()
+    ?.replace(/\.ts(x|)/g, "");
```

Comment:
```suggestion
    ?.replace(/\.tsx?/g, "");
```

Reply:
Renamed this to `fileName`. This function is actually used both on `sections` and `functions`